controllers/scopeinstance_controller.go

```diff
@@ -107,14 +209,92 @@ func (r *ScopeInstanceReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		}
 	}
 
-	log.Log.Info("No ScopeInstance error")
+	return nil
+}
 
-	return ctrl.Result{}, nil
+func (r *ScopeInstanceReconciler) deleteUnusedRBAC(ctx context.Context, clusterRoleBindingMap map[string]rbacv1.ClusterRoleBinding, roleBindingMap map[string]rbacv1.RoleBinding) error {
+
+	deleteObjects := []client.Object{}
+
+	// below is for cleanup, check if Map(Key, value) of Role Binding is empty or not if not then those
+	// are remaining ones which we have to delete
+	if len(roleBindingMap) != 0 {
+		for key, element := range roleBindingMap {
+			split := strings.Split(key, "-")
+
+			deleteObjects = append(deleteObjects, &rbacv1.RoleBinding{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      element.GetName(),
+					Namespace: split[0],
+					Labels: map[string]string{
+						"scopeInstance": "operators.io.operator-framework-v1",
+					},
+				},
+			})
+		}
+	}
+
+	// below is for cleanup, check if Map(Key, value) of Cluster Role Binding is empty or not if not then those
+	// are remaining ones which we have to delete
+	if len(clusterRoleBindingMap) != 0 {
```

Comment:
This `if` isn't needed. The for loop will be skipped if the length of the array that for loop is iterating over it.

Reply:
Applies in both places.

Reply:
there is no need to check. for loop will skip it.